moveit_ros/moveit_servo/test/pose_tracking_test.cpp

```diff
@@ -62,49 +64,92 @@ class PoseTrackingFixture : public ::testing::Test
 public:
   void SetUp() override
   {
-    // Wait for several key topics / parameters
-    ros::topic::waitForMessage<sensor_msgs::JointState>("/joint_states");
-    while (!nh_.hasParam("/robot_description") && ros::ok())
+    // Wait for several key topics / parameters // nodes to start
+    rclcpp::sleep_for(std::chrono::seconds(5));
+
+    executor_->add_node(node_);
+    executor_thread_ = std::thread([this]() { this->executor_->spin(); });
+    
+    if (!moveit_servo::readParameters(parameters_, node_, LOGGER))
     {
-      ros::Duration(0.1).sleep();
+      RCLCPP_ERROR_STREAM(LOGGER, "Could not get server parameters!");
+      exit(EXIT_FAILURE);
     }
 
     // Load the planning scene monitor
-    planning_scene_monitor_ = std::make_shared<planning_scene_monitor::PlanningSceneMonitor>("robot_description");
+     if (!planning_scene_monitor_->getPlanningScene())
+    {
+      RCLCPP_ERROR_STREAM(LOGGER, "Error in setting up the PlanningSceneMonitor.");
+      exit(EXIT_FAILURE);
+    }
+    
+    planning_scene_monitor_->providePlanningSceneService();
     planning_scene_monitor_->startSceneMonitor();
-    planning_scene_monitor_->startStateMonitor();
     planning_scene_monitor_->startWorldGeometryMonitor(
         planning_scene_monitor::PlanningSceneMonitor::DEFAULT_COLLISION_OBJECT_TOPIC,
         planning_scene_monitor::PlanningSceneMonitor::DEFAULT_PLANNING_SCENE_WORLD_TOPIC,
         false /* skip octomap monitor */);
+    planning_scene_monitor_->startStateMonitor(parameters_->joint_topic);
+    planning_scene_monitor_->startPublishingPlanningScene(planning_scene_monitor::PlanningSceneMonitor::UPDATE_SCENE);
 
-    tracker_ = std::make_shared<moveit_servo::PoseTracking>(nh_, planning_scene_monitor_);
-
-    target_pose_pub_ = nh_.advertise<geometry_msgs::PoseStamped>("target_pose", 1 /* queue */, true /* latch */);
+    rclcpp::sleep_for(std::chrono::seconds(2));
```

Comment:
Do we need this sleep?  What is it waiting for?

Reply:
I initially put it to set up the planning scene monitor but it's not necessary. I'll remove it